web/src/buddy_data.ts

```diff
@@ -347,7 +347,7 @@ function maybe_shrink_list(
 
     // We want to always show PM recipients even if they're inactive.
     const pm_ids_set = narrow_state.pm_ids_set();
-    const stream_id = narrow_state.stream_id();
+    const stream_id = narrow_state.stream_sub()?.stream_id;
```

Comment:
This looks correct, but I think someone will feel tempted to refactor this back to the original code. I'm not sure if there's a good way to define an abstraction, like `narrow_state.have_metadata_stream_id`, for this?

@evykassirer should probably review this commit.

Reply:
Can we instead have an optional param passed to `narrow_state.stream_id` like `only_valid_id` or something? I think it would be nicer to make it really clear what `stream_id` returns instead of sometimes calling one of these vs the other which look very similar.

Reply:
Okay. I went with @evykassirer's suggestion to add a boolean parameter to `narrow_state.stream_id` that defaults to `false` as that's the current behavior.